bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/MapUtils.java

```diff
@@ -86,7 +85,7 @@ public static void handleInherentTypeViolatingRecordUpdate(MapValue mapValue, BS
 
                 throw ErrorCreator.createError(
                         getModulePrefixedReason(MAP_LANG_LIB, INHERENT_TYPE_VIOLATION_ERROR_IDENTIFIER),
-                        BLangExceptionHelper.getErrorMessage(RuntimeErrors.RECORD_INVALID_READONLY_FIELD_UPDATE,
+                        BLangExceptionHelper.getErrorDetails(RuntimeErrors.RECORD_INVALID_READONLY_FIELD_UPDATE,
```

Comment:
Do we need this change since we keep the API to get BString and create details?

Reply:
We already have the [API](https://github.com/ballerina-platform/ballerina-lang/blob/f188b0bb552e70d58985097e927920d062278051/bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/creators/ErrorCreator.java#L126) which is getting error detail as `BMap<BString, Object>`. So we can use that to create internal runtime errors which are getting the detailed error message from `MessageBundle.properties` file. Just did a refactoring.